File: dotnet/ReportGenerator-VB/Robot.vb
Namespace ReportGeneratorVB
	Public Class Robot
        Private _bin As String
        Private _location As Machine

		Public Function Location() As Machine
            Return _location
		End Function

		Public Sub MoveTo(location As Machine)
            Me._location = location
		End Sub

		Public Sub Pick()
            _bin = _location.Take()
		End Sub

		Public Function Bin() As String
            Return _bin
		End Function

		Public Sub Release()
            _location.Put(Bin)
            _bin = Nothing
		End Sub
	End Class
End Namespace

File: production-line/csharp/ProductionLine/Robot.cs
namespace ProductionLine
{
    public class Robot
    {
        private Bin bin;
        private Machine location;

        public Machine Location()
        {
            return location;
        }

        public void MoveTo(Machine location)
        {
            this.location = location;
        }

        public void Pick()
        {
            bin = location.Take();
        }

        public Bin Bin()
        {
            return bin;
        }

        public void Release()
        {
            location.Put(bin);
            bin = null;
        }
    }
}